### ISSUE TITLE
Wrong value when flooring Y Component.

### ISSUE THREAD

**Issue by jonathansty** [redorav/hlslpp#9]
Below code snippet is the current behaviour for me. 
```
float1 y{ -0.01f };
float uf = hlslpp::floor(y); // returns -1 : ok
			
float3 broken{ -11.15f,-0.1f,-15.0f };
// Accessing the Y component is correct
float yVal = broken.y;
// next statement returns -12.0f -> Floor of x component
float actualf = hlslpp::floor(broken.y);
```
The floor function seems to be flooring my x component and returning that value instead of the Y component.

**Comment by redorav** [comment]
I see, there's an issue with some functions that return a floatN type instead of the swizzled components, which means the swizzle is lost through the function call. It affects other functions too, I'll take a careful look to fix them.

**Comment by redorav** [comment]
Fixed via 4430ace056655999da3c626de26c100e16d8f547
